sympy/core/tests/test_numbers.py

```diff
@@ -1749,3 +1749,41 @@ def test_Integer_precision():
     assert Float('1.0', precision=Integer(15))._prec == 15
     assert type(Float('1.0', precision=Integer(15))._prec) == int
     assert sympify(srepr(Float('1.0', precision=15))) == Float('1.0', precision=15)
+
+def test_numpy():
+    from sympy.utilities.pytest import skip
+    np = import_module('numpy')
+    def equal(x, y):
+        return x == y and type(x) == type(y)
+
+    if not np:
+        skip('numpy not installed.Abort numpy tests.')
+
+    assert Float(np.bool_(1)) is S(True)
+    try:
+        assert equal(
+            Float(np.int_(12345678912345678)), S(12345678912345678.))
+        assert equal(
+            Float(np.intp(12345678912345678)), S(12345678912345678.))
```

Comment:
A lot of these tests don't look right to me.  For example here, you can comparing the `S(<int>)` to `S(<float>)`.  They indeed could be the same but its not really what we're supposed to be testing.

Is this whole test a copy-paste with minor edits from sympify?  Maybe that is what we want but all those RHS with `.` are not right.

Reply:
These tests are indeed a copy-paste from test_sympify. Could you please clarify your comment - my motivation for adding `.` on the RHS was to make sure that the types agree with those on the LHS which all belong to `S(<float>)`.

Reply:
I agree with @Trave11er. The LHS is of type ```sympy.core.numbers.Float```, so for the RHS to have the same type, ```S()``` must take ```float``` as an input.

Reply:
As I tried to explain elsewhere, the RHS should be `Float('12345678912345678', something)`, for some value of `something`.  Bare Python `float`s should be avoided in these tests.

Reply:
Have rewritten the tests as you outlined in you comment with pseudocode below (to which I also can't reply weirdly). Have changed numbers.py to be the same as committed by @bmgxyz and it passes all the new tests (also, e.g. `Float(np.float64(5),'')` works now).